servicetalk-tcp-netty-internal/src/main/java/io/servicetalk/tcp/netty/internal/AbstractReadOnlyTcpConfig.java

```diff
@@ -51,8 +52,10 @@ protected AbstractReadOnlyTcpConfig(final AbstractTcpConfig<SecurityConfig, Read
         options = from.options() == null ? emptyMap() : unmodifiableMap(new HashMap<>(from.options()));
         idleTimeoutMs = from.idleTimeoutMs();
         flushStrategy = from.flushStrategy();
-        final String wireLoggerName = from.wireLoggerName();
-        wireLoggingInitializer = wireLoggerName != null ? new WireLoggingInitializer(wireLoggerName) : null;
+        final UserDataLoggerConfig wireLoggerConfig = from.wireLoggerConfig();
+        wireLoggingInitializer = wireLoggerConfig != null ?
+                new WireLoggingInitializer(wireLoggerConfig.loggerName(),
+                        wireLoggerConfig.logLevel(), wireLoggerConfig.logUserData()) : null;
```

Comment:
No idea why it was pre-existing, but it looks like we need just a `UserDataLoggerConfig` on the `AbstractReadOnlyTcpConfig` API. We should create `WireLoggingInitializer` inside `Tcp[Client|Server]ChannelInitializer` to make it consistent with all the other initializers we create there. `AbstractReadOnlyTcpConfig` should be just a config object without knowledge of initializers.